src/main/core/controller.c

```diff
@@ -465,6 +465,10 @@ gint controller_run(Controller* controller) {
                     controller->bootstrapEndTime, managerSeed, element->preloadPath.string->str,
                     element->environment.isSet ? element->environment.string->str : NULL);
 
+    if (controller->manager == NULL) {
```

Comment:
It's a bit surprising to have to check a `new` function for a NULL return. Since this is the only call site it's probably not that big of a deal in practice, but for an error that's ultimately going to result in exiting the simulation we might as well just `error` at that point rather than bubbling errors back up.

If we do want to make such a method fallible, we should probably call it something else; maybe `tryNew`

Reply:
I made all of the new code call error() instead of returning NULL, but I left the NULL check here since `manager_new()` can already return NULL from the existing code (since only a single manager can exist at a time), so I think it's still good to have this check.